bitcoin/src/blockdata/transaction.rs

```diff
@@ -1216,6 +1216,20 @@ where
     I: IntoIterator<Item = InputWeightPrediction>,
     O: IntoIterator<Item = usize>,
 {
+    // This fold() does three things.
+    // 1) counts the inputs and stores the sum as "input_counts".
+    //
+    // 2) Sums all of the input weights.  For every input:
+    //
+    //    script_size * 4 + witness_size
+    //
+    //    Sum the result of the above formula for every input.
+    //
+    //    Since script_size is non-witness data, it gets a 4x multiplier.
+    //    The results are stored in the variable "partial_input_weight".
+    //
+    // 3) counts the number of inputs that have a witness size greater than zero
+    //    and stores the count in the variable "inputs_with_witnesses".
```

Comment:
Couldn't help myself, made these changes while checking to see if I understood the code and docs. Feel free to use or not, all or any part of this:
```rust
    // This fold() does three things:
    // 1) Counts the inputs and returns the sum as `input_count`.
    // 2) Sums all of the input weights and return the sum as `partial_input_weight`
    //    For every input: script_size * 4 + witness_size
    //    Since script_size is non-witness data, it gets a 4x multiplier.
    // 3) Counts the number of inputs that have a witness data and returns the count as
    //    `num_inputs_with_witnesses`.
```

Reply:
oh, also I rekon `num_inputs_with_witnesses` is a better name but that is out of scope.